src/Tribe/DB_Lock.php

```diff
@@ -0,0 +1,355 @@
+		 * 1 row affected means we could INSERT and have the lock, 0 rows affected means we could not INSERT
+		 * and have not the lock.
+		 */
+
+		if ( $rows_affected ) {
+			self::$held_db_locks[ $lock_key ] = $lock_time;
+		}
+
+		return (bool) $rows_affected;
+	}
+
+	/**
+	 * Returns the option name used to manage the lock for a key in the options table.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $lock_key The lock key to build the option name for.
+	 *
+	 * @return string The name of the option that will be used to manage the lock for the specified key in the
+	 *                options table.
+	 */
+	public function get_db_lock_option_name( $lock_key ) {
+		return self::$db_lock_option_prefix . $lock_key;
+	}
+
+	/**
+	 * Releases the database lock of the record.
+	 *
+	 * Release a not held db lock will return `null`, not `false`.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $lock_key The name of the lock to release.
+	 *
+	 * @return bool Whether the lock was correctly released or not.
+	 */
+	public function release_db_lock( $lock_key ) {
+		if ( $this->manage_db_lock_w_mysql_functions() ) {
+			return $this->release_db_lock_w_mysql_functions( $lock_key );
+		}
+
+		return $this->release_db_lock_w_queries( $lock_key );
+	}
+
+	/**
+	 * Releases a DB lock held by the current database session (`$wpdb` instance) by
+	 * using the MySQL `RELEASE_LOCK` function.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $lock_key The lock key to release the lock for.
+	 *
+	 * @return bool Whether the lock was correctly released or not.
+	 */
+	protected function release_db_lock_w_mysql_functions( $lock_key ) {
+		global $wpdb;
+
+		$released = $wpdb->query(
+			$wpdb->prepare( "SELECT RELEASE_LOCK( SHA1( %s ) )", $lock_key )
+		);
+
+		if ( false === $released ) {
+			$log_data = [
+				'message' => 'Error while trying to release lock.',
+				'key'     => $lock_key,
+				'error'   => $wpdb->last_error
+			];
+			do_action( 'tribe_log', 'error', __CLASS__, $log_data );
+
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Releases a lock using SQL queries.
+	 *
+	 * Note: differently from the `release_db_lock_w_mysql_functions`, this method will release the lock
+	 * even if the current session is not the one holding the lock.
+	 * To protect from this the trait uses a map of registered locks and when the locks where registered.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $lock_key The lock key to release the lock for.
+	 *
+	 * @return bool Whether the lock was released or not, errors will be logged, a `false` value is returned if
+	 *              the lock was not held to begin with.
+	 */
+	protected function release_db_lock_w_queries( $lock_key ) {
+		if ( ! isset( self::$held_db_locks[ $lock_key ] ) ) {
+			// Avoid sessions that do nothold the lock to release it.
+			return false;
+		}
+
+		global $wpdb;
+		$option_name = $this->get_db_lock_option_name( $lock_key );
+		//phpcs:disable
```

Comment:
Why using PHPCS here?

Reply:
Direct calls to `DELETE` methods or calls to the `wpdb::delete` method should be accompained by cache invalidation.
If phpcs was not disabled here, the error would be reported.
There is no caching component being managed here, so I've suppressed the warnings.

Reply:
Uhn, interesting.